setup/setup.go

```diff
@@ -146,7 +148,10 @@ func StartManager(
 
 	namespace := os.Getenv("WATCH_NAMESPACE")
 	if namespace != "" {
+		setupLog.Info("WATCH_NAMESPACE env variable will be deprecated.\nPlease use the command line option -watch-namespace instead")
 		options.Namespace = namespace
+	} else if operatorOpts.WatchNamespace != "" {
+		options.Namespace = operatorOpts.WatchNamespace
```

Comment:
I think implementing this as a fallback is the better option e.g. if the cli argument is not set take the value from the env variable.